### ISSUE TITLE
(Unmanaged) Structs are not considered blittable by Emscripten / Blazor WASM AOT

### ISSUE THREAD

**Issue by HurricanKai** [dotnet/runtime#61146]
### Description

Unmanaged structs are not considered blittable and therefore cannot be used in pinvoke callbacks.

### Reproduction Steps

Attempt to compile code like
```cs
public struct S { public int Value; }

[UnmanagedCallersOnly]
public void M(S myStruct) { }
```
using Emscripten / WASM AOT

### Expected behavior

The above code should compile.

### Actual behavior

An error is shown, structs are not considered blittable.

### Configuration

Tested on .NET 6 RC2

**Comment by lambdageek** [comment]
This is an error from the AOT tasks, maybe?  Console sample using Mono works (see https://gist.github.com/lambdageek/f289259398a33cdf19811d2ab4eefd35):

```console
$ dotnet run
Hello, World! from Mono
Called with 42
```

**Comment by lambdageek** [comment]
Note that this is also related to https://github.com/dotnet/runtime/issues/60639 - in particular we will need to follow the runtime's (updated) rules for what is blittable

**Comment by lambdageek** [comment]
This could be considered an enhancement.

One current workaround is to pass a pointer to the struct:

```csharp

public struct S {
    public int i;
}


public class P {
    public static void Main () {
	unsafe {
	    Q (&M);
	} 
    }

    [UnmanagedCallersOnly]
    public unsafe static void M (IntPtr myStruct) {
	ref S s = ref Unsafe.AsRef<S>((void*)myStruct);
	Console.WriteLine ($"Called with {s.i}");
    }

    [DllImport("Foo")]
    public static extern unsafe void Q (delegate *unmanaged<IntPtr,void> fn);
}
```

And correspondingly use a pointer on the native side:
```c
typedef struct S {
	int i;
} S;

typedef void (*fn)(S* s);

void Q (fn f);

void Q (fn f) {
	S s;
	s.i = 42;
	f (&s);
}
```

**Comment by HurricanKai** [comment]
Given the workaround requires changes to the native side, which isn't always possible (ie binding to standardized or otherwise externally defined APIs), I would not consider this as an enhancement, but rather a missing feature that is important to making interop in WASM fully functional.

**Comment by ilonatommy** [comment]
> Note that this is also related to #60639 - in particular we will need to follow the runtime's (updated) per-assembly rules for what is blittable

The error still persists. Tried in Blazor wasm app with `RunAOTCompilation` and OP's code pasted to a class in Counter. Are we planning to fix it in 7.0?

**Comment by radical** [comment]
https://github.com/dotnet/runtime/blob/d24bf0991d02ab07b93b71edfed8634de818060c/src/tasks/WasmAppBuilder/PInvokeTableGenerator.cs#L344-L349

Assigning to @maraf as he was working on the table generation.
